src/spec-configuration/httpOCIRegistry.ts

```diff
@@ -213,13 +216,50 @@ async function getCredential(params: CommonParams, ociRef: OCIRef | OCICollectio
 		} catch (err) {
 			output.write(`[httpOci] Failed to read docker config.json: ${err}`, LogLevel.Trace);
 		}
+
+		if (!configContainsAuth) {
+			let defaultCredHelper = '';
+			// Try platform-specific default credential helper
+			if (process.platform === 'linux') {
+				if (existsInPath('pass')) {
+					defaultCredHelper = 'pass';
+				} else {
+					defaultCredHelper = 'secret';
+				}
+			} else if (process.platform === 'win32') {
+				defaultCredHelper = 'wincred';
+			} else if (process.platform === 'darwin') {
+				defaultCredHelper = 'osxkeychain';
+			}
+			if (defaultCredHelper !== '') {
+				output.write(`[httpOci] Invoking platform default credential helper '${defaultCredHelper}'`, LogLevel.Trace);
+				const auth = await getCredentialFromHelper(params, registry, defaultCredHelper);
+				if (auth) {
+					return auth;
+				}
+			}
+		}
 	}
 
 	// Represents anonymous access.
 	output.write(`[httpOci] No authentication credentials found for registry '${registry}'. Accessing anonymously.`, LogLevel.Trace);
 	return;
 }
 
+function existsInPath(filename: string): boolean {
+	if (!process.env.PATH) {
+		return false;
+	}
+	const paths = process.env.PATH.split(':');
+	for (const path of paths) {
+		const fullPath = `${path}/${filename}`;
+		if (fs.existsSync(fullPath)) {
```

Comment:
Could we use an async call to avoid blocking the JavaScript thread?

Reply:
you may consider taking a look at https://github.com/devcontainers/cli/blob/main/src/spec-utils/pfs.ts#L21-L31

Reply:
Updated to use `isLocalFile` and make this function async - thanks